src/ks_thread.c

```diff
@@ -389,7 +413,12 @@ static ks_status_t __init_os_thread(ks_thread_t *thread)
 	}
 #endif
 
+	ks_mutex_lock(thread->mutex);
+	thread->inuse = KS_TRUE;
+
 	if ((err = pthread_create(&thread->handle, &thread->attribute, thread_launch, thread)) != 0) {
+		thread->inuse = KS_FALSE;
+		ks_mutex_unlock(thread->mutex);
```

Comment:
This `in_use = false` and the unlock will make `thread` structure be possibly freed and therefore un-toucheable but it's used later in the code still.

Reply:
In the situation when end user runs ks_thread_create and ks_thread_destroy asynchronously, yes. If this is something possible, I can fix, np.

Reply:
Adding this I actually give the end user possiblity to call ks_thread_destroy async, which doesn't make sense, I'll change it

Reply:
Well. Looked again and I don't think that user knows the thread to call destroy at that point. So async is not possible.
I should take my words back.